### PR TITLE
Migrate 'Site Title block' e2e tests to Playwright

### PR DESCRIPTION
**Pull request by Mamaduka**
## What?
Part of #38851.

PR migrates 'Site Title block' e2e tests to Playwright.

## Why?
See #38851.

## Testing Instructions
```
npm run test:e2e:playwright -- test/e2e/specs/editor/blocks/site-title.spec.js
```

### CODE REVIEW

File: test/e2e/specs/editor/blocks/site-title.spec.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+		await editor.publishPost();
+		await page.reload();
+
+		await expect( siteTitleBlock ).toBeVisible();
+		await expect( siteTitleBlock ).toHaveText( 'New Site Title' );
+	} );
+
+	// Reason: Reason: The current e2e test setup doesn't provide an easy way to switch between user roles.
+	// eslint-disable-next-line playwright/expect-expect
+	test.fixme( 'Cannot edit the site title as editor', async () => {} );
```
Comment (t-hamano) [inline test/e2e/specs/editor/blocks/site-title.spec.js]: I tried code like below but it didn't work 😅 We can create a user, but it's difficult to switch to that user at the moment, right?

```javascript
test( 'Cannot edit the site title as editor', async ( {
	requestUtils,
	admin,
	page,
} ) => {
	await requestUtils.createUser( {
		username: 'testuser',
		email: 'testuser@example.com',
		firstName: 'Jane',
		lastName: 'Doe',
		password: 'secret',
	} );
	await page.reload();
	await admin.createNewPost();
	// ...
	await requestUtils.deleteAllUsers();
} );
```

Comment (Mamaduka) [reply test/e2e/specs/editor/blocks/site-title.spec.js]: That’s the reason this test is unfinished. We need to update our utility methods to support multiple roles.

**Review by t-hamano — APPROVED** [review]
LGTM 👍

Regarding the second test (`Cannot edit the site title as editor`), it was originally skipped, so I think there is no problem leaving it empty. In the future, when a utility method for switching users is added, we will be able to add an actual test.

File: test/e2e/specs/editor/blocks/site-title.spec.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+			.fill( 'New Site Title' );
+
+		await editor.publishPost();
+		await page.reload();
+
+		await expect( siteTitleBlock ).toBeVisible();
+		await expect( siteTitleBlock ).toHaveText( 'New Site Title' );
+	} );
+
+	// Reason: Reason: The current e2e test setup doesn't provide an easy way to switch between user roles.
```
Comment (t-hamano) [inline test/e2e/specs/editor/blocks/site-title.spec.js]:
```suggestion
	// Reason: The current e2e test setup doesn't provide an easy way to switch between user roles.
```

Nit

**Pull request merged**